sdk/python/feast/feature_store.py

```diff
@@ -998,6 +999,48 @@ def serve(self, port: int) -> None:
 
         feature_server.start_server(self, port)
 
+    @log_exceptions_and_usage
+    def upload_docker_image(self) -> None:
+        """Upload the docker image for the feature consumption server to the cloud."""
+
+        # TODO: add error checking and avoid hardcoding the region
+        repository_name = "feast-python-server-test"
+        feature_server_type = (
+            self.config.feature_server.type if self.config.feature_server else None
+        )
```

Comment:
You won't need to determine `feature_server_type` once the logic is part of AwsProvider. We'll just need to check whether feature server is enabled or not.

Reply:
Since we unified all provider logic in the PassthroughProvider in #1876, I think I'll have to continue checking for `feature_server_type`.